File: JavaScript/Source/RomanNumerals.js
class RomanNumber {
  constructor(letter, number) {
    this.letter = letter;
    this.number = number;
  }

  letter;

  number;
}

export class RomanNumerals {

  static #romanNumbers = [
    new RomanNumber('M', 1000),
    new RomanNumber('CM', 900),
    new RomanNumber('D', 500),
    new RomanNumber('CD', 400),
    new RomanNumber('C', 100),
    new RomanNumber('XC', 90),
    new RomanNumber('L', 50),
    new RomanNumber('XL', 40),
    new RomanNumber('X', 10),
    new RomanNumber('IX', 9),
    new RomanNumber('V', 5),
    new RomanNumber('IV', 4),
    new RomanNumber('I', 1),
  ];

  static fromRoman(roman) {
    let ret = 0;
    let i = 0;
    while (roman.length) {
      if (roman.startsWith(this.#romanNumbers[i].letter)) {
        ret += this.#romanNumbers[i].number;
        roman = roman.substring(this.#romanNumbers[i].letter.length);
      } else {
        i++;
      }
    }
    return ret;
  }

  static toRoman(number) {
    let i = 0;
    let ret = '';
    while (number) {
      if (number >= this.#romanNumbers[i].number) {
        ret += this.#romanNumbers[i].letter;
        number -= this.#romanNumbers[i].number;
      } else {
        i++;
      }
    }
    return ret;
  }
}

File: Python/Source/roman_numerals.py
# https://www.codewars.com/kata/51b66044bce5799a7f000003/csharp

from dataclasses import dataclass


@dataclass
class RomanNumber:
    letter: str
    number: int


class RomanNumerals:
    
    @staticmethod
    def to_roman(val):
        i = 0
        ret = ""
        while val > 0:
            if val >= RomanNumerals.__roman_numbers[i].number:
                ret = ret + RomanNumerals.__roman_numbers[i].letter
                val = val - RomanNumerals.__roman_numbers[i].number
            else:
                i = i + 1
        return ret
    
    @staticmethod
    def from_roman(roman_num):
        ret = 0
        i = 0
        while len(roman_num):
            if roman_num.startswith(RomanNumerals.__roman_numbers[i].letter):
                ret = ret + RomanNumerals.__roman_numbers[i].number
                roman_num = roman_num[len(RomanNumerals.__roman_numbers[i].letter):]
            else:
                i = i + 1
        return ret
    
    __roman_numbers = [
        RomanNumber("M", 1000),
        RomanNumber("CM", 900),
        RomanNumber("D", 500),
        RomanNumber("CD", 400),
        RomanNumber("C", 100),
        RomanNumber("XC", 90),
        RomanNumber("L", 50),
        RomanNumber("XL", 40),
        RomanNumber("X", 10),
        RomanNumber("IX", 9),
        RomanNumber("V", 5),
        RomanNumber("IV", 4),
        RomanNumber("I", 1),
    ]
